javascript/javascript3/week2/currency_converter/script.js

```diff
@@ -0,0 +1,82 @@
+"use strict";
+// initials
+const BASE_URL = "https://open.er-api.com";
+
+// elements
+const baseEl = document.getElementById("base");
+const targetEl = document.getElementById("target");
+const amountEl = document.getElementById("amount");
+const resultEl = document.getElementById("result");
+
+getRates(baseEl.value);
+
+/**
+ * Fetches currency rates for base currency, checks for errors and if status is ok,
+ * calls populateOptions - to display all the currencies in dropdowns and
+ * handleConvertion - to actually convert input value
+ * @param {string} base - currency code according to https://www.exchangerate-api.com/docs/supported-currencies
+ * @returns {object} of pairs [currency code]: [exchange rate to base currency]
+ */
+async function getRates(base) {
+  try {
+    const result = await fetch(`${BASE_URL}/v6/latest/${base}`);
+    if (result.status < 200 || result.status > 299) {
+      throw new Error(
+        `The request for currency rates was rejected with the code ${result.status}`
+      );
+    }
+    const currency = await result.json();
+
+    populateOptions(currency.rates, currency.base_code);
+
+    handleConvertion();
+    return currency.rates;
+  } catch (error) {
+    resultEl.innerText = `There was an error while fetching currencies... ${error.message}`;
```

Comment:
Which currency will be visible as `to currency` when the error is present?

Reply:
Great remark - I didn't have a default option for "to" field. Thanks!